Mathlib/MeasureTheory/Integral/SetIntegral.lean

```diff
@@ -403,6 +403,36 @@ theorem set_integral_eq_integral_of_forall_compl_eq_zero (h : ∀ x, x ∉ s →
   set_integral_eq_integral_of_ae_compl_eq_zero (eventually_of_forall h)
 #align measure_theory.set_integral_eq_integral_of_forall_compl_eq_zero MeasureTheory.set_integral_eq_integral_of_forall_compl_eq_zero
 
+lemma ae_restrict_eq_zero_iff_ae_eq_zero_of_mem {E : Type*} [Zero E] [MeasurableSpace E]
+    [MeasurableSingletonClass E] {f : α → E} {s : Set α}
+    (f_mble : NullMeasurable f (μ.restrict s)) :
+    f =ᵐ[Measure.restrict μ s] 0 ↔ ∀ᵐ x ∂μ, x ∈ s → f x = 0 := by
+  simp only [Measure.ae, MeasurableSet.compl_iff, EventuallyEq, Filter.Eventually,
+             Pi.zero_apply, Filter.mem_mk, mem_setOf_eq]
+  rw [Measure.restrict_apply₀]
+  · constructor <;> intro h <;> rw [← h] <;> congr <;> ext x <;> aesop
+  · apply NullMeasurableSet.compl
+    convert f_mble (MeasurableSet.singleton 0)
+
+lemma ae_restrict_eq_zero_iff_ae_eq_zero_of_mem' {E : Type*} [Zero E] (f : α → E) {s : Set α}
+    (s_mble : MeasurableSet s) :
+    f =ᵐ[Measure.restrict μ s] 0 ↔ ∀ᵐ x ∂μ, x ∈ s → f x = 0 := by
```

Comment:
could you replace zero here by an arbitrary constant in E (and remove the class `Zero E`)?